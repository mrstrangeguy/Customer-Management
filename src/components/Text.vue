<template>
  <div
    :class="{ 'h-8': true, 'flex items-center': icon }"
  >
    <div v-if="icon" class="p-1.5 flex items-center">
      <i
        :class="`default-icon block ${icon} h-4 w-4 text-center leading-4 text-[#0000001a]`"
      />
    </div>
    <span
      v-if="!icon"
      class="block text-[10px] leading-2.5 text-[#00000099]"
      >{{ title }}</span
    >
    <div
    style="color: descriptionColor;"
      :class="{
        [`text-xs leading-3.5 mt-1`]: true,
        'mt-0': icon,
        'py-[9px] leading=[15.4284px] px-3 text-black': icon,
      }"
    >
      {{ description }}
    </div>
  </div>
</template>

<script setup lang="ts">
defineProps({
  icon: String,
  title: String,
  description: String,
  descriptionColor: {
    type: String,
    default: "#000",
  },
});
</script>

<style scoped></style>

<!-- <Text title="Position" description="ACME" description-color="#03a9f4" />
<Text icon="dx-icon dx-icon-tel" description="sample5@gmail.com" /> -->
