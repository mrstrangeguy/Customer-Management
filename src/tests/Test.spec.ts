import { it, describe, expect } from "vitest";
import { shallowMount } from "@vue/test-utils";
import Text from "../components/IconTextField.vue";

describe("Text", () => {
  const primaryText = shallowMount(Text, {
    props: {
      title: "Position",
      descriptionColor: "#03a9f4",
      description: "This is description",
    },
  });

  const secondaryText = shallowMount(Text, {
    props: {
      icon: "dx-icon dx-icon-tel",
      description: "sample5@gmail.com",
    },
  });

  it("test snapshot of primary text component", () => {
    expect(primaryText.element).toMatchSnapshot();
  });

  it("test snapshot of secondary text component", () => {
    expect(secondaryText.element).toMatchSnapshot();
  });
});
