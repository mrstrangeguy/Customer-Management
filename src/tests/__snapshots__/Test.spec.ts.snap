// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Text > test snapshot of primary text component 1`] = `
<div
  class="h-8"
>
  <!--v-if-->
  <span
    class="block text-[10px] leading-2.5 text-[#00000099]"
  >
    Position
  </span>
  <div
    class="text-xs mt-1"
    style="color: rgb(3, 169, 244);"
  >
    This is description
  </div>
</div>
`;

exports[`Text > test snapshot of secondary text component 1`] = `
<div
  class="h-8 flex items-center"
>
  <div
    class="p-1.5 align-middle flex items-center"
  >
    <i
      class="default-icon block leading-[18px] h-4 w-4 text-center  text-[#0000001a] dx-icon dx-icon-tel"
    />
  </div>
  <!--v-if-->
  <div
    class="text-xs py-[9px] leading-[14px] px-3"
    style="color: rgb(0, 0, 0);"
  >
    sample5@gmail.com
  </div>
</div>
`;
